Loop/Managers/DeviceDataManager.swift

```diff
@@ -1151,12 +1162,15 @@ extension DeviceDataManager: LoopDataManagerDelegate {
             return units
         }
 
-        return pumpManager.roundToSupportedBolusVolume(units: units)
+        let rounded = ([0.0] + pumpManager.supportedBolusVolumes).enumerated().min( by: { abs($0.1 - units) < abs($1.1 - units) } )!.1
+        self.log.default("Rounded %{public}@ to %{public}@", String(describing: units), String(describing: rounded))
+
+        return rounded
```

Comment:
I don't understand why the pumpManager isn't responsible for this rounding. Why do we want to round in the DeviceDataManager?

Reply:
Also, if there is no pumpManager, why is `units` returned for a bolus that cannot be delivered? Could NaN (or similar) be used?

Reply:
1) Because how rounding is done is a Loop responsibility. Possible values are PumpManager responsibility, but if the dosing algorithm wants a value between possible values, Loop should decide which actual value is chosen.
2) Just a convenience to avoid having to deal with nils in display values, I'd guess, since this is also used for that. Could be changed if you feel strongly about it.